### ISSUE TITLE
How to define my own Preprocessing function in Python when using NNEstimator?

### ISSUE THREAD

**Issue by anyuecq25** [intel-analytics/analytics-zoo#4101]
How to define my own Preprocessing function in Python when using NNEstimator?

As depicted in this url.
https://analytics-zoo.readthedocs.io/en/latest/doc/UseCase/nnframes.html

NNEstimator(model, criterion, featurePreprocessing: Preprocessing[F, Tensor[T]], labelPreprocessing: Preprocessing[F, Tensor[T]])

Takes model, criterion, featurePreprocessing and labelPreprocessing. NNEstimator will extract the data from feature and label columns and convert each feature/label to Tensor with the featurePreprocessing and labelPreprocessing. This constructor provides more flexibility in supporting extra data types.

**Comment by qiuxin2012** [comment]
It's a little difficult to define your own preprocessing function for python user. Because our preprecessing is running in scala actually. 
Here is a pull request #681 about how to add a preprocessing in Analytics Zoo.
Be free to ask for our help if you have difficulty writing scala code. And welcome your contribute if your preprocessing is a common case.

**Comment by jason-dai** [comment]
@anyuecq25 Alternatively, you may use the new Orca API (e.g., see https://analytics-zoo.readthedocs.io/en/latest/doc/Orca/QuickStart/orca-keras-quickstart.html), where you can use standard TensorFlow Dataset or PyTorch DataLoader for your preprocessing.

**Comment by anyuecq25** [comment]
> It's a little difficult to define your own preprocessing function for python user. Because our preprecessing is running in scala actually.
> Here is a pull request #681 about how to add a preprocessing in Analytics Zoo.
> Be free to ask for our help if you have difficulty writing scala code. And welcome your contribute if your preprocessing is a common case.

Thanks!

**Comment by anyuecq25** [comment]
> @anyuecq25 Alternatively, you may use the new Orca API (e.g., see https://analytics-zoo.readthedocs.io/en/latest/doc/Orca/QuickStart/orca-keras-quickstart.html), where you can use standard TensorFlow Dataset or PyTorch DataLoader for your preprocessing.

Thanks!
